tests/ctst/common/common.ts

```diff
@@ -357,32 +359,37 @@ Then('kafka consumed messages should not take too much place on disk', { timeout
 
                 for (let i = 0; i < topics.length; i++) {
                     this.logger.debug('Checking topic', { topic: topics[i] });
+                    let topicCleaned = false;
                     for (let j = 0; j < newOffsets[i].partitions.length; j++) {
-                        const newMessagesAfterClean =
-                            newOffsets[i].partitions[j].low === previousOffsets[i].partitions[j].high &&
-                            newOffsets[i].partitions[j].high > previousOffsets[i].partitions[j].high;
+                        const newOffsetPartition = newOffsets[i].partitions[j];
+                        const oldOffsetPartition = previousOffsets[i].partitions[j];
 
-                        if (newMessagesAfterClean) {
-                            // If new messages appeared after we gathered the offsets, we need to recheck after
-                            this.logger.warn('New messages after clean', { topic: topics[i] });
+                        if (!oldOffsetPartition) {
                             continue;
                         }
 
-                        const lowOffsetIncreased = newOffsets[i].partitions[j].low >
-                            previousOffsets[i].partitions[j].low;
-                        const allMessagesCleaned = newOffsets[i].partitions[j].high ===
-                            newOffsets[i].partitions[j].low;
-
-                        // If the low offset increased it means the topic has been cleaned
-                        // If low offset is the same as high offset,
-                        // it means the topic is completly cleaned even though lowOffset didnt increased
-                        assert.ok(lowOffsetIncreased || allMessagesCleaned,
-                            `Topic ${topics[i]} partition ${j} offset has not increased,
-                            previousOffsets: ${previousOffsets[i].partitions[j].low} /\
-                            ${previousOffsets[i].partitions[j].high},
-                            newOffsets: ${newOffsets[i].partitions[j].low} / ${newOffsets[i].partitions[j].high}`);
-
-                        // Topic is cleaned, we don't need to check it anymore
+                        // Ensure we're accessing the correct partition details
+                        const lowOffsetIncreased = parseInt(newOffsetPartition.low) >
+                            parseInt(oldOffsetPartition.low);
+                        // We tolerate one message not being cleaned, as it can be due to the
+                        // message being consumed during the check
+                        const allMessagesCleaned = parseInt(newOffsetPartition.low) + 1 >=
+                            parseInt(newOffsetPartition.high);
+
+                        // We consider one topic as xleaned if kafkacleaner affected the
```

Comment:
Will be done before merging, keeping the PR as-is for now, for the review.

```suggestion
                        // We consider one topic as cleaned if kafkacleaner affected the
```